qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -539,26 +590,31 @@ def compute_minimum_eigenvalue(
 
         start_time = time()
 
-        # keep this until Optimizer.optimize is removed
-        try:
-            opt_result = self.optimizer.minimize(
+        if callable(self.optimizer):
+            opt_result = self.optimizer(  # pylint: disable=not-callable
                 fun=energy_evaluation, x0=initial_point, jac=gradient, bounds=bounds
             )
```

Comment:
About the type hint: yeah, this doesn't call things positionally.  I think you should note clearly in the documentation that you'll be passing things with named keyword arguments.

Reply:
I'm resolving this, referring to the above comment on the typehint 🙂